torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -464,6 +464,60 @@ Val* IterDomain::extent() const {
   return extent_;
 }
 
+namespace {
+
+class RejectMultipleGridReductions : public IterVisitor {
+ public:
+  static void analyze(Fusion* fusion) {
+    RejectMultipleGridReductions multi_grid;
+    multi_grid.traverse(fusion, true);
+  }
+
+ private:
+  void handle(ReductionOp* rop) override {
+    TensorView* out = dynamic_cast<TensorView*>(rop->out());
```

Comment:
just a suggestion - for conditional casts, I found the pattern bellow to be a bit more explicit and also more compact:

```cpp
if (auto out = dynamic_cast<TensorView*>(rop->out()) {
  ... do stuff with out ...
}
```

Reply:
I don't have a strong opinion, but generally, I like to minimize conditional nesting. E.g., instead of:

```
if (X) {
  if (Y) { 
    if (Z) {
       do_something;
     }
   }
}
```

I'd prefer:
```
if (!X) return;
if (!Y) return;
if (!Z) return;
do_something;
```

Reply:
I agree that excessive nesting is a code smell (that shouldn't be the case here). The reason I prefer the if (cast) { ... } is that it minimizes, and makes the scope of the cast explicit.

Anyway, it's just a small suggestion, no objections if you decide to keep the code as is.